src/Application.vala

```diff
@@ -103,12 +103,54 @@ public class Music.Application : Gtk.Application {
         settings.bind ("window-maximized", main_window, "maximized", SettingsBindFlags.SET);
     }
 
+    File[] listDirectory(string directory) {
+        var dir = Dir.open(directory, 0);
+
+        string? name = null;
+        File[] elements = {};
+
+        while((name = dir.read_name ()) != null) {
+            string filePath = Path.build_filename (directory, name);
+            File file = File.new_for_path(filePath);
+
+            elements += file;
+        }
+
+        return elements;
+    }
+
+    private bool isDirectory(string file) {
+        return FileUtils.test(file, FileTest.IS_DIR);
+    }
+
+    private void loopThroughFiles(File[] files) {
+        // All of these are added later in bulk
+        File[] elements = {};
+
+        foreach (var file in files) {
+            var filePath = file.get_path();
+            var isDirectory = isDirectory(filePath);
+
+            if (!isDirectory) {
+                elements += file;
+
+                continue;
+            }
+
+            // Is a directory
+            var directoryElements = listDirectory(filePath);
+            loopThroughFiles(directoryElements);
+        }
+
+        playback_manager.queue_files (elements);
```

Comment:
Nice work, this is better now then adding files one by one. But i think it would be even better if `loop_through_files` will return array of files and then they will be added to queue in `open` method.

Reply:
Is there a way to move the elements of an array to another in vala? E.g. move `[1, 2, 3]` to array `[4, 5, 6]`. Result: `[4, 5, 6, 1, 2, 3]` not: `[4, 5, 6, [1, 2, 3]]`

Reply:
You can add them in a loop

Reply:
What do you mean? Check my new code at [line 136](https://github.com/elementary/music/pull/747/commits/a049712dd1446e63fb68a74cf89afaeea49bb8d9#diff-f4e5d48d751c670a8575715da1cb0a0976532427a8b6143c7fdede415a33ba6cR136)

Reply:
Oh it's fine now, sorry

Reply:
You need to look through loop_through_files again and add elements one by one